enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -1075,29 +1107,38 @@ func (c *FirecrackerContainer) Create(ctx context.Context, actionWorkingDir stri
 	if err != nil {
 		return status.InternalErrorf("Failed creating machine: %s", err)
 	}
-	if err := m.Start(vmCtx); err != nil {
+	if err := c.startMachine(ctx, vmCtx, m); err != nil {
 		return status.InternalErrorf("Failed starting machine: %s", err)
 	}
 	c.machine = m
 	return nil
 }
 
+func (c *FirecrackerContainer) startMachine(ctx context.Context, startCtx context.Context, m *fcclient.Machine) error {
```

Comment:
might be clearer to not put this in a separate method like this. The two contexts coming from somewhere else are a bit mind boggling.

(also, not relevant if you remove this, but when you have more than one sequential arguments to a function in go that have the same type, you can omit the type information for all but the last one to save space)

Reply:
Moved the tracing back to the original callsite, with the help of a util func that wraps an anonymous func with a span. wdyt about this approach?

Reply:
Resolved offline, using an inline closure instead